manual/manual/cmds/intf-c.etex

```diff
@@ -986,15 +986,42 @@ Use the normal C array syntax instead.
 with the garbage collector using the "caml_register_global_root" function.
 \end{gcrule}
 
-Registration of a global variable "v" is achieved by calling
+The one may need to access values in OCaml heap from external code, for
+example when main loop of the program is implemented in C++. In this case the
+value of type "value" can be stored outside the OCaml heap for later use. In this
+subsection these external locations will be called \emph{global} values.
+
+Saving an OCaml value outside OCaml heap in a global value will work
+well when a piece of data is an unboxed integer and may work well when the it is a pointer
+outside OCaml heap. However, if a piece of data is a \emph{pointer to the OCaml heap} the
+one needs to register another location of this data in the OCaml runtime.
+Registration of a global value "v" of type "value" is achieved by calling
 "caml_register_global_root(&v)" just before or just after a valid
-value is stored in "v" for the first time. You must not call any
-of the OCaml runtime functions or macros between registering and
-storing the value.
+value is stored in "v" for the first time. No any of the OCaml runtime functions
+or macros should be called between registering and storing the value.
 
-A registered global variable "v" can be un-registered by calling
+A registered global value "v" can be un-registered by calling
 "caml_remove_global_root(&v)".
 
+The importance of registering global values consists of two aspects. The first
+is that data being registered may be sweeped by OCaml garbage collector. In that
+case it should be explicitly explained
+to the GC that this data should be \emph{alive} as far as program runs
+(or until it will be unregistered using the function above),
+i.e. this value should be a \emph{root} in the graph of references between
+all OCaml values.
+
+The second aspect is related to a number of heaps in the OCaml runtime. There are
+two generations: the minor heaps for ``young'' values and the major one
+for ``old'' ones. When a ``young'' value becomes old enough it is promoted to the
+major heap and moved from old memory address to another one. If this
+value was registered as a global root the OCaml runtime should
+guarantee that global value still points to the right
+data (holds new pointer). This issue is resolved by the function "caml_register_global_root" which
```

Comment:
The sentence "This issue is resolved by the function …" introduces a strange repetition since we are already inside a paragraph explaning why `caml_register_global_root` is needed. If I am not mistaken, the following paragraph is explaining why the first argument to `caml_register_global_root` is a pointer to a value and not a value.